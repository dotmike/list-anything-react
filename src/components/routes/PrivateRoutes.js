import React from "react";
import PropTypes from "prop-types";

import UserRoute from "../routes/UserRoute";
import MainContentContainer from "../containers/MainContentContainer";

import DashboardPage from "../pages/DashboardPage";
import NewSubjectPage from "../pages/NewSubjectPage";
import SubjectPage from "../pages/SubjectPage";
import NewSubjectDataPage from "../pages/NewSubjectDataPage";
import SubjectDataPage from "../pages/SubjectDataPage";
import SearchPage from "../pages/SearchPage";

const PrivateRoutes = ({ location, history, showContent }) => (
  <MainContentContainer history={history} showContent={showContent}>
    <UserRoute
      location={location}
      path="/dashboard"
      exact
      component={DashboardPage}
    />
    <UserRoute
      location={location}
      path="/subjects/new"
      exact
      component={NewSubjectPage}
    />
    <UserRoute
      location={location}
      path="/subject/:_id"
      exact
      component={SubjectPage}
    />
    <UserRoute
      location={location}
      path="/subject/:_id/add"
      exact
      component={NewSubjectDataPage}
    />
    <UserRoute
      location={location}
      path="/subject-data/:_id"
      exact
      component={SubjectDataPage}
    />
    <UserRoute
      location={location}
      path="/search"
      exact
      component={SearchPage}
    />
  </MainContentContainer>
);

PrivateRoutes.propTypes = {
  // ownProps
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired,
  showContent: PropTypes.bool.isRequired
};

export default PrivateRoutes;
