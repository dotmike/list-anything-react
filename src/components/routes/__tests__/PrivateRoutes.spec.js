import React from "react";

import PrivateRoutes from "../PrivateRoutes";

const defaultProps = {
  history: {
    push: () => {},
    location: { pathname: "/dashboard" }
  },
  location: { pathname: "/dashboard" },
  showContent: false
};

describe("PrivateRoutes", () => {
  it("renders correctly", () => {
    const wrapper = shallowWithContext(<PrivateRoutes {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
