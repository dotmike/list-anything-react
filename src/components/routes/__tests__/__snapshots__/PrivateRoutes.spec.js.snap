// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoutes renders correctly 1`] = `
<MainContentContainer
  history={
    Object {
      "location": Object {
        "pathname": "/dashboard",
      },
      "push": [Function],
    }
  }
  showContent={false}
>
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/dashboard"
  />
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/subjects/new"
  />
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/subject/:_id"
  />
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/subject/:_id/add"
  />
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/subject-data/:_id"
  />
  <Connect(UserRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/dashboard",
      }
    }
    path="/search"
  />
</MainContentContainer>
`;
